PolyEngine/RenderingDevice/OpenGL/Src/BlinnPhongRenderingPass.cpp

```diff
@@ -104,6 +118,27 @@ void BlinnPhongRenderingPass::OnRun(World* world, const CameraComponent* camera,
 	}
 	GetProgram().SetUniform("uPointLightCount", pointLightsCount);
 
+	int spotLightsCount = 0;
+	for (auto componentsTuple : world->IterateComponents<SpotLightComponent, TransformComponent>())
+	{
+		SpotLightComponent* spotLightCmp = std::get<SpotLightComponent*>(componentsTuple);
+		TransformComponent* transformCmp = std::get<TransformComponent*>(componentsTuple);
+
+		String baseName = String("uSpotLight[") + String::From(spotLightsCount) + String("].");
+		GetProgram().SetUniform(baseName + "Range", spotLightCmp->GetRange());
+		GetProgram().SetUniform(baseName + "CutOff", Cos(1.0_deg * spotLightCmp->GetCutOff()));
+		GetProgram().SetUniform(baseName + "OuterCutOff", Cos(1.0_deg * spotLightCmp->GetOuterCutOff()));
+		GetProgram().SetUniform(baseName + "Position", transformCmp->GetGlobalTranslation());
+		GetProgram().SetUniform(baseName + "Direction", MovementSystem::GetGlobalForward(transformCmp));
+		GetProgram().SetUniform(baseName + "Base.Color", spotLightCmp->GetColor());
+		GetProgram().SetUniform(baseName + "Base.Intensity", spotLightCmp->GetIntensity());
+
+		++spotLightsCount;
+		if (spotLightsCount == MAX_LIGHT_COUNT_SPOT)
```

Comment:
We can provide API in `ShaderProgram` that enables seting shader constnts to certain value from C++ code. Sounds good? ;)

Reply:
yes